src/pollution_model/run_medusa_2.py

```diff
@@ -28,33 +30,161 @@
+        # Store the building_point_data to the database table
+        log.info("Adding roof_surface_points table to the database.")
+        roof_surface_points.to_postgis("roof_surface_points", engine, index=False, if_exists="replace")
+
+
+def read_roof_surface_points(engine: Engine) -> gpd.GeoDataFrame:
+    """
+    Read building data under points. Then store them into database.
+
+    Parameters
+    ----------
+    engine : Engine
+        The engine used to connect to the database.
+    """
+    # Check if the table already exist in the database
+    if tables.check_table_exists(engine, "roof_surface_polygons"):
+        log.info("roof_surface_polygons data already exists in the database.")
+    else:
+        # Read roof surface polygons from outside
+        roof_surface_polygons = gpd.read_file(
+            "/home/martinnguyen204/Digital-Twin/polygons_fullversion.shp"
+        )
+        # Rename building_Id to building_id
+        roof_surface_polygons = roof_surface_polygons.rename(columns={
+            'building_I': 'building_id',
+            'Shape_Leng': 'Shape_Length'
+        })
+        # Store the building_point_data to the database table
+        log.info("Adding roof_surface_polygons table to the database.")
+        roof_surface_polygons.to_postgis("roof_surface_polygons", engine, index=False, if_exists="replace")
+
+
+def merge_roof_surface(engine: Engine) -> gpd.GeoDataFrame:
+    """
+    Merge building data under polygons and points. Then store them into database.
+
+    Parameters
+    ----------
+    engine : Engine
+        The engine used to connect to the database.
+    """
+    # Read building data under polygon format
+    read_roof_surface_polygons(engine)
+
+    # Read building data under point format
+    read_roof_surface_points(engine)
+
+    # Check if the table already exist in the database
+    if tables.check_table_exists(engine, "roof_surface"):
+        log.info("roof_surface data already exists in the database.")
+    else:
+        # Merge building points and polygons using inner join,
+        # because there will be some building_id in polygon table
+        # that do not have roof surface type
+        query = """
+            SELECT roof_surface_polygons.*, roof_surface_points.deeplearn_subclass
+            FROM roof_surface_polygons
+            INNER JOIN roof_surface_points
+            USING ("building_id")
+        """
+        roof_surface_merge = gpd.GeoDataFrame.from_postgis(query, engine, geom_col="geometry")
+
+        # Add column surface type
+        roof_surface_types = {
+            'ColourSteel': 'Cs',
+            'MetalTile': 'Mt',
+            'Galvanised': 'Gv',
+            'NonMetal': 'NoMt',
+            'MetalOther': 'Mo',
+            'Zincalume': 'ZnC',
+            'AsphaltRoad': 'Rd',
+            'CarPark': 'CrP'
+        }
+        roof_surface_merge['surface_type'] = roof_surface_merge.deeplearn_subclass.map(roof_surface_types)
```

Comment:
I think we should not change this if possible. I will come back to look into this

Reply:
In fact, I don't think we should save this query as a separate table. It would be better to only have the original tables and query on the fly. Its not an expensive query and reduces database complexity. What are your thoughts?